src/main/java/com/minecolonies/api/colony/buildingextensions/IBuildingExtension.java

```diff
@@ -108,4 +108,10 @@ public interface IBuildingExtension extends IModuleContainer<IBuildingExtensionM
      * Equals implementation for this building extension.
      */
     boolean equals(Object other);
+
+    /**
+     * Get the unique extension id.
+     * @return the unique id.
+     */
+    int getId();
```

Comment:
I think position makes a better identity than an int we need to keep unique, why not use the blockpos? Its similar to the buildings and building manager